--- pkg-plist.orig	2011-08-24 00:30:55.682683562 -0400
+++ pkg-plist	2011-08-24 00:31:13.236692521 -0400
@@ -262,9 +262,9 @@
 include/gtk-unix-print-2.0/gtk/gtkprintjob.h
 include/gtk-unix-print-2.0/gtk/gtkprintunixdialog.h
 include/gtk-unix-print-2.0/gtk/gtkunixprint.h
-lib/girepository-1.0/Gdk-2.0.typelib
-lib/girepository-1.0/GdkX11-2.0.typelib
-lib/girepository-1.0/Gtk-2.0.typelib
+%%INTROSPECTION%%lib/girepository-1.0/Gdk-2.0.typelib
+%%INTROSPECTION%%lib/girepository-1.0/GdkX11-2.0.typelib
+%%INTROSPECTION%%lib/girepository-1.0/Gtk-2.0.typelib
 lib/gtk-2.0/%%GTK2_VERSION%%/engines/libpixmap.a
 lib/gtk-2.0/%%GTK2_VERSION%%/engines/libpixmap.la
 lib/gtk-2.0/%%GTK2_VERSION%%/engines/libpixmap.so
@@ -335,9 +335,9 @@
 libdata/pkgconfig/gtk+-unix-print-2.0.pc
 libdata/pkgconfig/gtk+-x11-2.0.pc
 share/aclocal/gtk-2.0.m4
-share/gir-1.0/Gdk-2.0.gir
-share/gir-1.0/GdkX11-2.0.gir
-share/gir-1.0/Gtk-2.0.gir
+%%INTROSPECTION%%share/gir-1.0/Gdk-2.0.gir
+%%INTROSPECTION%%share/gir-1.0/GdkX11-2.0.gir
+%%INTROSPECTION%%share/gir-1.0/Gtk-2.0.gir
 share/gtk-2.0/demo/alphatest.png
 share/gtk-2.0/demo/apple-red.png
 share/gtk-2.0/demo/appwindow.c
--- Makefile.orig	2011-08-24 00:30:14.755663229 -0400
+++ Makefile	2011-08-24 00:40:40.394973532 -0400
@@ -32,7 +32,6 @@
 
 .if !defined(REFERENCE_PORT)
 
-BUILD_DEPENDS+=	g-ir-scanner:${PORTSDIR}/devel/gobject-introspection
 RUN_DEPENDS+=	${LOCALBASE}/share/mime/magic:${PORTSDIR}/misc/shared-mime-info \
 		${LOCALBASE}/share/icons/hicolor/index.theme:${PORTSDIR}/misc/hicolor-icon-theme
 
@@ -56,7 +55,8 @@
 CONFIGURE_ARGS+=	--enable-debug=yes
 .endif
 
-OPTIONS=	CUPS "Enable cups printing support" on
+OPTIONS=	CUPS "Enable cups printing support" on \
+		INTROSPECTION "Enable introspection" on
 
 .include <bsd.port.options.mk>
 
@@ -69,6 +69,15 @@
 PLIST_SUB+=	CUPS:="@comment "
 .endif
 
+.if defined(WITH_INTROSPECTION)
+BUILD_DEPENDS+= g-ir-scanner:${PORTSDIR}/devel/gobject-introspection
+CONFIGURE_ARGS+=--enable-introspection=yes
+PLIST_SUB+=	INTROSPECTION=""
+.else
+CONFIGURE_ARGS+=--enable-introspection=no
+PLIST_SUB+=	INTROSPECTION="@comment "
+.endif
+
 post-patch:
 	@${FIND} ${WRKSRC} -name "Makefile.in" | ${XARGS} ${GREP} -l lgmodule | \
 		${XARGS} ${REINPLACE_CMD} -e 's|-lgmodule|@GMODULE_LIBS@|g'
