--- src/callbacks.c.dist	2006-12-19 05:54:45.000000000 -0500
+++ src/callbacks.c	2006-12-19 06:07:04.000000000 -0500
@@ -598,9 +598,9 @@
 	pref = (gtkTermPref *) user_data;
 	notebook = pref->notebook;
 
-	if (event->state & GDK_SHIFT_MASK)
+	if (event->state & GDK_CONTROL_MASK)
 	{
-		if ((event->keyval == GDK_Right) || (event->keyval == GDK_KP_Right))
+		if (event->keyval == GDK_Page_Down)
 		{
 			cur_term = gtk_notebook_get_current_page (GTK_NOTEBOOK(notebook));
 			gtk_notebook_next_page(GTK_NOTEBOOK(notebook));
@@ -616,7 +616,7 @@
 			return TRUE;
 		}
 
-		else if ((event->keyval == GDK_Left) || (event->keyval == GDK_KP_Left))
+		else if (event->keyval == GDK_Page_Up)
 		{
 			cur_term = gtk_notebook_get_current_page (GTK_NOTEBOOK(notebook));
 			if (cur_term == 0)
@@ -633,10 +633,8 @@
 
 			return TRUE;
 		}
-	}
-	if (event->state & GDK_CONTROL_MASK)
-	{
-		if (event->keyval == GDK_n)
+
+		else if (event->keyval == GDK_t)
 		{
 			on_new_tab_activate (GTK_MENU_ITEM(lookup_widget(GTK_WIDGET(fenster), "new_tab")), pref);
 
