diff -urN /main/pports/mail/claws-mail/Makefile /usr/ports/mail/claws-mail/Makefile
--- /main/pports/mail/claws-mail/Makefile	2012-12-10 12:32:44.000000000 -0500
+++ /usr/ports/mail/claws-mail/Makefile	2013-01-19 17:54:30.246421150 -0500
@@ -18,18 +18,7 @@
 		cairo:${PORTSDIR}/graphics/cairo \
 		png15:${PORTSDIR}/graphics/png \
 		etpan:${PORTSDIR}/mail/libetpan \
-		gmp:${PORTSDIR}/math/gmp \
-		freetype:${PORTSDIR}/print/freetype2 \
-		sasl2:${PORTSDIR}/security/cyrus-sasl2 \
-		gnutls:${PORTSDIR}/security/gnutls \
-		gcrypt:${PORTSDIR}/security/libgcrypt \
-		gpg-error:${PORTSDIR}/security/libgpg-error \
-		tasn1:${PORTSDIR}/security/libtasn1 \
-		nettle:${PORTSDIR}/security/nettle \
-		p11-kit:${PORTSDIR}/security/p11-kit \
-		expat:${PORTSDIR}/textproc/expat2 \
-		fontconfig:${PORTSDIR}/x11-fonts/fontconfig \
-		xcb:${PORTSDIR}/x11/libxcb
+		gnutls:${PORTSDIR}/security/gnutls
 RUN_DEPENDS=	${LOCALBASE}/etc/mime.types:${PORTSDIR}/misc/mime-support
 
 CONFLICTS=	sylpheed-claws
@@ -39,7 +28,7 @@
 USE_ICONV=	yes
 USE_XORG=	ice pixman sm x11 xau xcomposite xcursor xdamage xdmcp xext \
 		xfixes xi xinerama xrandr xrender
-USE_GNOME=	atk glib20 gnomehack gtk20 libartlgpl2 libxml2 pango
+USE_GNOME=	atk glib20 gnomehack gtk20 libxml2 pango
 GNU_CONFIGURE=	yes
 USE_PKGCONFIG=	build
 INSTALLS_ICONS=	yes
